### PR TITLE
fix: fix 2d renderer order error

### PR DESCRIPTION
**Pull request by singlecoder**
### Please check if the PR fulfills these requirements

- [ ] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Review by GuoLei1990 — APPROVED** [review]
+1

**Review by GuoLei1990 — APPROVED** [review]
+1

File: packages/core/src/2d/sprite/SpriteRenderer.ts [hunk 1]
```diff
@@ -18,6 +18,13 @@ export class SpriteRenderer extends Renderer {
   private static _textureProperty: ShaderProperty = Shader.getPropertyByName("u_spriteTexture");
   private static _tempVec3: Vector3 = new Vector3();
 
+  /** @internal temp solution. */
+  @ignoreClone
+  _customLocalBounds: BoundingBox = null;
+  /** @internal temp solution. */
+  @ignoreClone
+  _curtomRootEntity: Entity = null;
```
Comment (cptbtptpbcptdtptp) [inline packages/core/src/2d/sprite/SpriteRenderer.ts:26]: _customRootEntity

**Review by cptbtptpbcptdtptp — CHANGES_REQUESTED** [review]

File: packages/core/src/RenderPipeline/RenderQueue.ts [hunk 2]
```diff
@@ -19,23 +19,33 @@ export class RenderQueue {
...
    */
   static _compareFromFarToNear(a: Item, b: Item): number {
     const renderQueueDif = a.material.renderQueueType - b.material.renderQueueType;
-
     if (renderQueueDif) {
       return renderQueueDif;
     }
-    return b.component._distanceForSort - a.component._distanceForSort;
+
+    const distanceDif =  b.component._distanceForSort - a.component._distanceForSort;
```
Comment (eyworldwide) [inline packages/core/src/RenderPipeline/RenderQueue.ts:43]:
```typescript
return a.material.renderQueueType - b.material.renderQueueType || 
b.component._distanceForSort - a.component._distanceForSort ||
 b.component._renderSortId - a.component._renderSortId;
```

File: packages/core/src/2d/sprite/SpriteRenderer.ts (hunk 1 above)
Comment (singlecoder) [reply packages/core/src/2d/sprite/SpriteRenderer.ts]: > _customRootEntity

Done

**Review by eyworldwide — APPROVED** [review]
+1

**Pull request merged**